azure-functions/acbs-function/helpers/mandatoryFields.js

```diff
@@ -1,5 +1,12 @@
-const findMissingMandatory = (data, mandatoryFields) =>
-  mandatoryFields.map((mf) => (data[mf] !== null || data[mf] !== undefined || data[mf] !== '' ? '' : mf)).filter((mf) => mf !== '');
+/**
+ * Finds missing mandatory fields in the provided data object.
+ *
+ * @param {Object} data - The data object to check for missing fields.
+ * @param {Array<string>} mandatoryFields - An array of field names that are mandatory.
+ * @returns {Array<string>} - An array of missing mandatory field names. If all mandatory fields are present, returns an empty array.
+ */
+
+const findMissingMandatory = (data, mandatoryFields) => mandatoryFields.map((field) => (data[field] ? false : field)).filter((field) => field);
```

Comment:
not sure what others think, worth splitting this into a 2 functions so easier to read/understand?

Reply:
IMO something like this would be easier to read

```ts
const mapMissingField = (field) => data[field] ? false : field;

const findMissingMandatoryFields = (data, mandatoryFields) => {
  const mapped = mandatoryFields.map((field) => mapMissingField(field))

  // I am not sure why we need this (?)
  const filtered = mapped.filter((field) => field);

  return filtered;
};
```

Reply:
Bump @abhi-markan